src/metrics/family.rs

```diff
@@ -107,7 +107,17 @@ pub struct Family<S, M> {
     /// [`Histogram`](crate::metrics::histogram::Histogram) in order to set
     /// specific buckets, a custom constructor is set via
     /// [`Family::new_with_constructor`].
-    constructor: fn() -> M,
+    constructor: C,
+}
+
+pub trait MetricConstructor<M> {
```

Comment:
Would you mind adding some documentation pointing to `Family::new_with_constructor` and a doc example to this type?

Reply:
I added docs and an example to the trait, and also to the `Fn() -> M` implementor for completeness. Thanks for the suggestion